### PR TITLE
shylubasker: fix shylubasker transpose solve correctness in amesos2

### PR DESCRIPTION
**Pull request by ndellingwood**
Added example to test through Amesos2

## Testing
Tested in local build to confirm changes address transpose solve correctness issues when calling shylubasker through amesos2

### CODE REVIEW

**Comment by ndellingwood** [comment]
@hkthorn this is one more small PR to fix correctness of transpose solves when calling shylubasker through amesos2. This PR does not include a couple other issues you've pointed out that need addressing (I wanted to fix transpose solve correctness first with the solver), a follow on PR will address the inconsistency of setting transpose mode in Amesos2, and remove need to create different solver instances for basker in cases when a user will want to compute forward and transpose solves.

File: packages/amesos2/example/SimpleSolveTranspose.cpp [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+
+/**
```
Comment (srajama1) [inline packages/amesos2/example/SimpleSolveTranspose.cpp]: Cut and paste, may be ?

File: packages/amesos2/example/SimpleSolveTranspose.cpp [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+  typedef Tpetra::CrsMatrix<Scalar,LO,GO> MAT;
+  typedef Tpetra::MultiVector<Scalar,LO,GO> MV;
+
+  using Tpetra::global_size_t;
+  using Teuchos::tuple;
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+
+
+  // Before we do anything, check that SuperLU is enabled
```
Comment (srajama1) [inline packages/amesos2/example/SimpleSolveTranspose.cpp]: Another cut and paste ?

File: packages/amesos2/example/SimpleSolveTranspose.cpp (hunk 1 above)
Comment (ndellingwood) [reply packages/amesos2/example/SimpleSolveTranspose.cpp]: Yes, cleaned up in newest commit I just pushed, good catch.

File: packages/amesos2/example/SimpleSolveTranspose.cpp (hunk 2 above)
Comment (ndellingwood) [reply packages/amesos2/example/SimpleSolveTranspose.cpp]: Yes, just cleaned this up in recent commit and cleaned up the commit history.

**Comment by ndellingwood** [comment]
PR failed because I pushed changes while the autotester was running, next run through should be clean.

**Comment by ndellingwood** [comment]
Needed to namespace a use of "endl" in the example I added to amesos2, not sure why it built fine for me locally. Commit with fix pushed, retesting.

**Comment by ndellingwood** [comment]
Thanks @hkthorn !

**Pull request merged**